bvm/ballerina-core/src/main/java/org/ballerinalang/core/model/types/BUnionType.java

```diff
@@ -91,8 +92,27 @@ public int getTag() {
 
     @Override
     public String toString() {
-        List<String> list = memberTypes.stream().map(BType::toString).collect(Collectors.toList());
-        return String.join("|", list);
+        String cachedToString;
+        if (resolving) {
+            if (this.typeName != null) {
+                return this.typeName;
+            } else {
+                return "...";
```

Comment:
Can `this.typeName` be null when `resolving` is true? Wouldn't `resolving` be true only for a cyclic union, which in turn has to have a name?

Reply:
Not necessarily if someone creates one programmatically.